src/ttauri/version.hpp.in

```diff
@@ -0,0 +1,41 @@
+// Copyright Take Vos 2020.
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <string>
+
+namespace tt {
+
+struct version {
+    /** Incremented on backward incompatible change. */
+    int major = @TTAURI_MAJOR_VERSION@;
```

Comment:
I would like these to be set at initialization point of the ttauri_version.

Otherwise when the application needs a git_version it will hold these values, I rather have them not filled in yet.

Reply:
I don't understand. Please explain this concept. 
Are you refering to the runtime of the application here? Do you want to dynamically load the version data from outside?

Normally, the version strings are set either as #defines or to a struct/class with access helper methods to the version header file. 

We talk about versioning the ttauri-library here, right? An application based on TT has it's own `version.h`.

Reply:
The application will need to pass its version information to ttauri, so that is why I want to have this as a generic struct.

Some subsystems, specifically right now Vulkan requires the application name and version, together with the engine name ("ttaur") and version. Presumably for the vulkan driver to recognize a specific application/game and engine to have different settings or bug-for-bug compatibility.

Maybe I would like to have the name of the application in this same struct. Maybe it shouldn't be called `version`.

Reply:
vulkan, requires just the name as string and semantic major,minor,patch-version values as integers.

Reply:
> Maybe it shouldn't be called version.

This sounds more like a collection of meta data collected for vulkan, e.g. `version_info_for_vulkan` or `metadata_for_vulkan`.

> The application will need to pass its version information to ttauri, so that is why I want to have this as a generic struct.

My perspective is more from the app side: the application gets the library information. 
Think of an application with an about box displaying: `MyCalculator v1.2.3 using ttauri v3.2.1`.

Reply:
So, instead of having all kinds of meta data for each sub system. Lets make meta information complete enough so that we can display a generic about dialogue, together with initializing subsystems. And for determining all the information needed to rebuild the application.